sdk/personalizer/ai-personalizer-rest/test/public/configurationTests.spec.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Recorder, delay, env } from "@azure-tools/test-recorder";
+import { assert } from "chai";
+import { createRecorder } from "./utils/recordedClient";
+import { Context } from "mocha";
+import createPersonalizerClient, {
+  PersonalizerClient,
+  PolicyContract,
+  PolicyContractOutput,
+  PolicyUpdateParameters,
+  ServiceConfiguration,
+  ServiceConfigurationOutput,
+  ServiceConfigurationUpdateParameters,
+  isUnexpected,
+} from "../../src";
+
+describe("Configuration Tests", () => {
+  let recorder: Recorder;
+  let client: PersonalizerClient;
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    client = createPersonalizerClient(
+      env["PERSONALIZER_ENDPOINT_SINGLE_SLOT"] ?? "",
+      {
+        key: env["PERSONALIZER_API_KEY_SINGLE_SLOT"] ?? "",
+      },
+      recorder.configureClientOptions({})
+    );
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("update configuration test", async function () {
+    const configuration: ServiceConfigurationUpdateParameters = {
+      body: {
+        rewardAggregation: "average",
+        modelExportFrequency: "PT3M",
+        defaultReward: 1.0,
+        rewardWaitTime: "PT4H",
+        explorationPercentage: 0.3,
+        logRetentionDays: 100,
+        learningMode: "Online",
+      },
+    };
+
+    const updatedConfiguration = await updateConfigurationAsync(client, configuration);
+    assertServiceConfigurationEquals(updatedConfiguration, configuration.body);
+    await delay(30 * 1000);
```

Comment:
We should be careful here and make sure we mock the clock otherwise we'll actually be waiting for 3 seconds.

Please take a look at Sinon's faketimer api: https://sinonjs.org/releases/latest/fake-timers/ we have a few tests using it in the repo.

We can't have tests using delay

Reply:
The delay function is imported from "@azure-tools/test-recorder". So when I run in playback mode, there is no delay at all.

Reply:
Can you wap the delay in a playback check to be extra safe?

```ts
if (!isPlayback()){
   await delay(30 * 1000);
}
```

Reply:
Though we won't wait in playback, it is a bad design to wait for 30 seconds in general.
If the service needs us to wait for sometime, it should be part of the API, for instance, it could be a long running operation leveraging lro's.

For now, we can overlook it, but eventually, we should find ways to get rid of the delays.

Reply:
Sounds good. We have a feature underway that has sticky sessions that will allow us to remove these waits once that feature is live.